tests/model/test_rnn.py

```diff
@@ -84,9 +89,179 @@ def test_correct_rnn_type_and_size_returned(
         bias += getattr(rnn.rnn, f"bias_hh_l{l}")[
             hidden_size : 2 * hidden_size
         ]
+        bias = torch.tensor(bias).cpu()
         assert torch.allclose(bias, torch.tensor(forget_gate_bias))
 
 
+@given(
+    rnn_type=rnn_types(),
+    input_size=st.integers(min_value=1, max_value=8),
+    hidden_size=st.integers(min_value=1, max_value=8),
+    num_layers=st.integers(min_value=1, max_value=3),
+    bias=st.booleans(),
+    dropout=st.floats(min_value=0.0, max_value=1.0, allow_nan=False),
+    bidirectional=st.booleans(),
+    forget_gate_bias=st.one_of(
+        st.none(), st.floats(min_value=-10.0, max_value=10.0, allow_nan=False)
+    ),
+    batch_first=st.booleans(),
+    batch_size=st.integers(min_value=1, max_value=4),
+    max_seq_len=st.integers(min_value=1, max_value=8),
+)
+@settings(deadline=4000)
+def test_rnn_forward_pass_no_hidden(
```

Comment:
The `FullyConnected` layer tests have a `def fully_connecteds(...) -> st.SearchStrategy[Union[FullyConnected, Tuple[FullyConnected, Dict]]]` function that produces a valid `FullyConnected` layer and optionally returns the dictionary of `kwargs` used to create it. 

It then uses this, combined with: https://github.com/MyrtleSoftware/myrtlespeech/blob/fd88be189c686e8940243d7c548a8255e83b3f7d/tests/utils/utils.py#L82 

to easily test that the module runs. 

Can this setup be replicated here? It vastly simplifies the actual tests and moves the logic of creating valid RNNs and tensors out of the test itself (and the section that has a `deadline`!). 

The `RNNState` could be generated by another `SearchStrategy` too meaning both tests would merge together.

Reply:
Done - I've refactored the data generation and deleted the (now repeating) test. 

I'm using the `.../utils/utils.py:: tensors()` to generate the rnn input but was not able to find an easy way to do this for the `RNNState` since it's subtype and dimensions are v strongly coupled with the `RNNType`, rnn `kwargs` and input data `batch_size`.